### PR TITLE
Update Python and dependency versions

### PR DESCRIPTION
**Pull request by NirantK**
1. Adds support for Python 3.12 in dependencies
2. Upgrades ONNX and ONNX-Runtime in order to do so
3. Upgrades other dependencies to be as latest as possible
4. Adds Python 3.12 tests for 3 OS: mac, windows, linux - latest series
5. New Poetry lock file

### CODE REVIEW

**Comment by NirantK** [comment]
Fixes #21

File: pyproject.toml [hunk 1]
```diff
@@ -11,14 +11,18 @@ repository = "https://github.com/qdrant/fastembed"
...
-tqdm = "^4.65"
+python = ">=3.8.0,<3.13"
+onnx = "^1.15.0"
+onnxruntime = "^1.17.0"
+tqdm = "^4.66"
 requests = "^2.31"
-tokenizers = "^0.15.0"
-huggingface-hub = "0.19.4"
+tokenizers = "^0.15.1"
+huggingface-hub = "^0.20.3"
```
Comment (joein) [inline pyproject.toml:20]: Do we need to be this strict here?

Why not 

```
tokenizers = "^0.15"
huggingface-hub = "^0.20"
```

Comment (NirantK) [reply pyproject.toml:20]: It's the later versions which have been released and tested for 3.12 

e.g. in the case of tokenizers: This PR is part of the latest release and adds Windows tests for Python 3.12: https://github.com/huggingface/tokenizers/pull/1438/

I'd also prefer to more relaxed but don't want this to break for a specific dependency.

Comment (NirantK) [inline pyproject.toml:20]:
```suggestion
huggingface-hub = "^0.20"
```

Relax version lock for Huggingface Hub

Comment (joein) [reply pyproject.toml]: So the minimal version which we support is 0.15.1?

Comment (joein) [reply pyproject.toml]: And our options will be either unlock the dependency with `^0.16` once it is released or replace it with `>=0.15.1,<1.0.0`

**Review by joein — APPROVED** [review]

**Pull request merged**